modules/gapi/src/backends/common/gbackend.hpp

```diff
@@ -58,6 +58,35 @@ namespace gimpl {
     struct Data;
     struct RcDesc;
 
+    struct RMatMediaBGRAdapter final: public cv::RMat::Adapter
+    {
+        RMatMediaBGRAdapter(cv::MediaFrame frame) : m_frame(frame) { };
+
+        virtual cv::RMat::View access(cv::RMat::Access a) override
+        {
+            auto view = m_frame.access(a == cv::RMat::Access::W ? cv::MediaFrame::Access::W
+                                                                : cv::MediaFrame::Access::R);
+            auto ptr = reinterpret_cast<uchar*>(view.ptr[0]);
+            auto stride = view.stride[0];
+
+            std::shared_ptr<cv::MediaFrame::View> view_ptr =
+                std::make_shared<cv::MediaFrame::View>(std::move(view));
+            auto callback = [view_ptr]() mutable { view_ptr.reset(); };
```

Comment:
@rgarnov Did you try to use `unique_ptr` here instead of shared?

Reply:
I believe `unique_ptr` can't be _copied_ to a lambda's associated captured scope. Not sure if it can be moved there, either.

Reply:
Right, C++14 is required for that ("move" in lambda capturing).